XenAdmin/Diagnostics/Problems/HostProblem/HostPrepareToUpgradeProblem.cs

```diff
@@ -0,0 +1,97 @@
+﻿/* Copyright (c) Citrix Systems, Inc. 
+ * All rights reserved. 
+ * 
+ * Redistribution and use in source and binary forms, 
+ * with or without modification, are permitted provided 
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+
+using System.Drawing;
+using XenAdmin.Core;
+using XenAdmin.Diagnostics.Checks;
+using XenAdmin.Dialogs;
+using XenAPI;
+
+namespace XenAdmin.Diagnostics.Problems.HostProblem
+{
+    class HostPrepareToUpgradeProblem : HostProblem
+    {
+        private readonly string _shortMessage;
+        private readonly string _longMessage;
+
+        public HostPrepareToUpgradeProblem(Check check, Host host, string friendlyErrorKey = null)
+            : base(check, host)
+        {
+            InitialiseMessagesFromErrorKey(friendlyErrorKey, out _shortMessage, out _longMessage);
+        }
+
+        public override bool IsFixable => false;

```

Comment:
One thing I'd like to draw the attention to here is IsFixable being set to false: this results to the Resolve All button being disabled. Strictly speaking this is not wrong - we cannot resolve _all_ since not all are fixable, some have to be manually fixed. It poses the question though whether we might want to change Resolve All to mean resolve all fixable.